Newer musl expands NULL to nullptr, which is not a bool.
Either way, the check is always true with openssl 1.1+, so it can probably removed alltogether
---
--- mumble-1.4.287/src/SSL.cpp	2022-09-13 19:24:40.000000000 +0200
+++ -	2023-02-07 22:18:16.093759105 +0100
@@ -33,7 +34,7 @@
 	// If we detect that no locking callback is configured, we
 	// have to set it up ourselves to allow multi-threaded use
 	// of OpenSSL.
-	if (!CRYPTO_get_locking_callback()) {
+	if (CRYPTO_get_locking_callback() == NULL) {
 		SSLLocks::initialize();
 	}
 }
